lib/api/streaming.js

```diff
@@ -127,12 +139,17 @@ Streaming.prototype._createClient = function(replay) {
 };
 
 /** @private **/
-Streaming.prototype._getFayeClient = function(channelName) {
+Streaming.prototype._getFayeClient = function(channelName, extension) {
   var isGeneric = channelName.indexOf('/u/') === 0;
   var clientType = isGeneric ? 'generic' : 'pushTopic';
   if (!this._fayeClients || !this._fayeClients[clientType]) {
     this._fayeClients = this._fayeClients || {};
     this._fayeClients[clientType] = this._createClient(isGeneric);
+
+    if (typeof extension !== "undefined") {
+      this._fayeClients[clientType].addExtension(extension)
+    }
+
```

Comment:
Implicitly modifies faye client with addExtension, which leads the following normal getFayeClient(channelName) calls to return the client instance with extension.
Should be much explicit API interface if you are going to modify the client behavior.

Reply:
I have the opposite view, I think extensions should just be an added optional parameter when creating a Topic. The faye client is specific to the Topic so adding extensions to it should be treated the same as the channel name.